# Biased methods

Flux Balance Analysis
---------------------

Among the biased methods, the most well-known technique is flux balance
analysis (FBA) which uses the assignment of stoichiometric coefficients
in a matrix to represent the metabolites involved in any given reaction
in a metabolic network [@orth2010flux]. Through these coefficients,
constraints can be imposed on the system to identify all potential flux
distributions which are associated with a corresponding set of feasible
phenotypic states. The aim of FBA is to locate a value (or set of
values) in the solution space which best satisfies a given objective
function. FBA uses linear programming to solve this objective function,
indicating the extent to which each reaction in the network contributes
to a phenotypic state. The most common objective function computed by
FBA is the synthesis of biomass, which is commonly used to indicate
cellular growth rate and predict product yields [@feist2010biomass].
Fluxes can either be calculated under the steady state assumption, which
postulates that the total amount of any metabolite being produced must
be equal to the total amount of that metabolite consumed
[@zielinski2016metabolic] or in a dynamic state where changes in
specific concentrations and kinetics parameters have been recorded for
each metabolite over time (e.g for DFBA) [@mahadevan2002dynamic].
Experimental validation of model predictions for DFBA are often obtained
from 13C metabolic flux analysis [@wiechert200113], which utilises
isotopic-labeling of metabolic substrates to quantify intracellular
fluxes. Additionally, methods such as dynamic multi-species metabolic
modelling (DMMM) have been used to examine inter-species competition for
metabolites in a microbial community [@zhuang2011genome].

Numerous modifications of FBA propose the application of various
constraints to shrink the solution space for determining the precise
flux state of the cell by calculating the optimal set of solutions for a
given objective via linear programming. In most instances, constraints
are defined by cell and reaction stoichiometry, fluxes through transport
and metabolic reactions, upper and lower bounds for each flux, biomass
composition and ATP requirements [@reed2012shrinking]. Upper and lower
bounds can be estimated using flux variability analysis, which returns
the maximum and minimum fluxes through each reaction whilst maintaining
minimal biomass production [@mahadevan2003effects].

Linear thermodynamic constraints are applied in thermodynamic metabolic
flux analysis (TMFA) and thermodynamic variability analysis to eliminate
thermodynamically infeasible reactions or loops from pathways and gather
information on feasible metabolite activity and Gibbs free energy
changes [@henry2007thermodynamics]. The removal of thermodynamically
infeasible loops is necessary to prevent violating the loop law, which
states that there is no net flux through balanced biochemical loops in
networks at steady state [@price2002extreme]. Loopless COBRA methods
solve a modified mixed-integer problem with the added constraint of no
network fluxes containing loops; application of this constraint has been
described for FBA, FVA and MCMC sampling
[@schellenberger2011elimination]. Fast flux variability analysis with
thermodynamic constraints (tFVA) removes unbounded fluxes from
biochemical loops arising from non-zero, steady-state fluxes involving
internal reactions [@muller2013fast]. This is a faster implementation of
FVA which does not require the specification of metabolite
concentrations or additional experimental data, although these have been
included in other variants of the method.

Parsimonious FBA (pFBA) identifies a subset of genes contributing to
maximising the growth rate *in-silico*, therefore enabling maximisation
of stoichiometric efficiency [@lewis2010omic]. Conditional dependencies
present in the metabolic model can be applied as constraints for each
flux, whereby each flux is constrained by the activity of the compound
that facilitates it. This technique is known as conditional FBA and has
proved to be effective for simulating phototrophic growth and diurnal
dynamics in cyanobacteria [@rugen2015elucidating]
[@reimers2016evaluating]. Resource balance analysis (RBA) uses growth
rate limitation caused by distribution of proteins between cellular
processes to constrain flux predictions [@goelzer2011bacterial].
Although more time-consuming than a linear programming approach,
Bayesian flux estimation results in a probability density function which
is more stable and informative than a simple point estimate
[@heino2007bayesian]. The METABOLICA statistical framework utilises a
Bayesian approach to performing FBA. Metabolism is modelled in a
multi-compartment macroscopic model with a stochastic extension of the
stationary state and the Bayesian inference problem is solved by
computing posterior probability densities using MCMC sampling
[@heino2010metabolica].

Alternatively, standard constraints may be removed to construct new FBA
methods to improve flux predictions for non-steady-state or
non-wild-type cells. Relaxing the assumption of fixed reactant
proportions for biomass production is the basis of flexible FBA
(flexFBA), which can be coupled with relaxing the fixed ratio of
byproduct to reactant (known as time-linked FBA) to observe transitions
between steady states [@birch2014incorporation]. Combining these methods
also enables the comparison of metabolite production between knockout
mutants.

To further increase the informativity of constraint-based models, the
integration of information from regulatory pathways and external
multi-omic data are described in the following section.

Regulatory Methods
------------------

Steady-state regulatory flux balance analysis (SR-FBA) is used to
quantify the extent to which metabolic and transcriptional regulatory
constraints affect the state of flux activity for various metabolic
genes [@shlomi2007genome]. SR-FBA allows for improved characterisation
of steady-state metabolic behaviour compared to regulatory FBA (rFBA),
which chooses a single steady state per time interval from all possible
solutions to find the flux distribution consistent with the regulatory
state of each interval. Integrated FBA (iFBA) incorporates metabolic,
regulatory and signaling pathways in the FBA model to enable thorough
characterisation of dynamic-state metabolic behaviour
[@covert2008integrating]. Integrated dynamic flux balance analysis
(idFBA) additionally couples fast and slow reactions to give
quantitative time-variant flux predictions [@lee2008dynamic]. However,
all of these approaches are limited by the Boolean logic formalism,
which restricts the definition of gene activity to an on/off state. This
disadvantage can be overcome by using conditional probabilities to
represent gene states and gene-transcription factor interactions when
combining high throughput data with regulatory networks, as demonstrated
by the probabilistic regulation of metabolism (PROM) method
[@chandrasekaran2010probabilistic]. This approach allows for a greater
number of interactions between metabolic models and their respective
transcriptional regulatory networks to be recorded as they are
quantified automatically [@chandrasekaran2013metabolic].

The generation of condition-specific metabolic models utilises the
integration of transcriptomic profiles to detect gene activity across
conditions and may be divided into two main approaches: switch-based and
valve based methods. Switch-based algorithms remove inactive or lowly
expressed genes by setting the corresponding reaction boundaries to zero
before FBA is performed [@salehzadeh2014computational]. Gene inactivity
moderated by metabolism and expression (GIMME) finds a flux distribution
optimising a given objective and avoids the use of so-called ’inactive’
reactions which are below a certain transcription threshold
[@vivek2016advances] [@kim2014methods]. The main advantage of GIMME is
that it can re-enable flux associated with false negative values in
inactive reactions and record consistency between gene expression data
and the predicted flux distribution for a given objective
[@becker2008context].

Tissue-specific gene and protein expression values can be integrated
into genome scale metabolic models accounting for different metabolic
objectives at the cellular level [@shlomi2008network], in order to
extract information regarding the uptake and secretion of metabolites by
specific tissue and cell-types. For this, tissue-specific variations in
enzyme expression levels are used to inform the likelihood of enzymes
supporting flux in their associated reactions by categorising
gene-to-reaction mapping for each reaction in the model corresponding to
the level of gene expression (i.e. high, moderate or low expression).
Subsequently, fluxes corresponding to high gene expression are maximised
and those corresponding to low gene expression are minimised when
solving a mixed-integer linear program [@blazier2012integration]. This
process has been developed into the Integrative Metabolic Analysis Tool
(iMAT) [@zur2010imat] which displays the most likely predicted metabolic
fluxes corresponding to reactions in metabolic models. This tool enables
the definition of a biological objective to be dependent on the
requirements of each cell rather than the entire organism. An extension
of iMAT called EXAMO enables the design of condition-specific metabolic
models for human tissues [@rossell2013inferring]. Similarly, the
integrative network inference for tissues (INIT) algorithm uses
tissue-specific information collected from the Human Protein Atlas to
help incorporate transcriptomic and proteomic data into a genome-scale
model and produce cell-type specific metabolic networks
[@agren2012reconstruction]. This data forms the input for a
mixed-integer linear problem which modifies the steady-state condition
by setting a small positive net accumulation rate for internal
metabolites [@machado2014systematic]. Net production of these
metabolites are assigned positive weights, corresponding to arbitrary
scores for the level of protein expression. The algorithm for metabolic
adjustment by differential expression (MADE) compares the fold changes
of gene expression values between conditions to intuitively predict the
most consistent and statistically-significant metabolic adjustments
[@jensen2011functional]. The fold changes are expressed as a series of
binary expression states, for which differences between successive
states most closely mirror corresponding differences in the mean
expression levels.

On the other hand, valve-based algorithms reduce the activity of lowly
expressed genes by adjusting the upper and lower bounds for their
corresponding reactions proportional to the normalised expression of the
genes associated with those reactions before performing FBA
[@salehzadeh2014computational]. Such methods include E-flux
[@colijn2009interpreting], METRADE [@angione2015predictive], FALCON
[@barker2015robust] and PROM [@chandrasekaran2010probabilistic]. For
valve-based methods, gene expression data is not discretised as in
switch-based methods. There are also methods which treat gene expression
data as absolute as opposed to relative, meaning that these data are
more indicative of protein concentrations as levels of transcription are
more comparable across genes [@lee2012improving]. In E-flux, flux
boundaries are tightly constrained when gene expression is low but
relaxed when gene expression is high [@colijn2009interpreting];
transcript levels can be used to set an upper bound for the maximum
production of enzymes and therefore constrain all reaction rates
[@kim2014methods]. FALCON is a novel algorithm which estimates enzyme
abundances using gene-protein-reaction (GPR) rules in the model, thus
improving the predictive capability of models integrated with expression
data [@barker2015robust].

Transcriptional-controlled FBA (tFBA) uses constraints between pairs of
conditions based on gene expression data for the optimisation of FBA,
considering both fold change and absolute change in expression to
minimise noise [@van2011predicting]. Recently, a new method has been
introduced for incorporating expression data as well as transcriptional
regulatory networks to simulate growth under various environmental and
genetic perturbations [@motamedian2017trfba]. Transcriptional regulated
flux balance analysis (TRFBA) applies two unique linear constraints.
Firstly, reaction rates are limited using a constant which sets
expression levels equal to the upper bounds of reactions; secondly, the
expression level of each gene is correlated with the genes they are
regulated by. One important advantage of TRFBA is the ability to improve
predictions of growth without requiring detailed information about
transcriptional regulators and their target genes. A novel method known
as omFBA [@guo2016om] uses a phenotype-match algorithm to formulate the
optimal objective function i.e. that which yields the most accurate
estimations of the observed phenotypes. This objective can be
simultaneously correlated with multiple omic data types via regression
analysis to generate a omics-guided objective function, consequently
resulting in a clearer correlation between genotype and phenotype and
improved phenotypic predictions.

Gene perturbation and objective function selection
--------------------------------------------------

Deciding upon the number of gene perturbations to be performed depends
on multiple factors. Choosing to perform single or pairwise gene
perturbations one-by-one may fail to capture the essentiality and
function of that gene as a result of genetic redundancy (i.e. there may
be multiple genes encoding the same function). However, concurrently
knocking out multiple genes can cause issues related to scaling unless
coupled with e.g. Shapley value analysis which assigns a contribution
value to each gene knockout in the system [@deutscher2008can]. Synthetic
lethality can be described as the simultaneous inactivation of a pair of
non-essential genes resulting in the death of a cell or organism
[@nijman2011synthetic]. Knocking out multiple synthetic lethal pairs for
genome-scale metabolic models can help in analysing the structural
robustness of metabolic networks and identifying interdependencies among
genes and reactions [@suthers2009genome].

Flux ratios can be applied as constraints for FBA using the flux balance
analysis with flux ratios (FBrAtio) algorithm, which can be directly
implemented into the stoichiometric matrix of genome-scale metabolic
models [@mcanulty2012genome] [@yen2013deriving]. In this algorithm,
multiple enzymes compete for metabolic branch points in the network
(known as critical nodes) [@mcanulty2012genome] which specify how a
substrate in the metabolite pool is distributed between competing
reactions; this depends on factors relating to thermodynamics such as
enzyme availability and downstream accumulation of reactive
intermediates. The optimisation of flux ratios for a particular
phenotype can be achieved through partial knockdown, overexpression or
total knockout of enzyme-coding genes [@mcanulty2012genome]. As opposed
to complete knockouts, performing gene over-expression or partial
knockdown may prove to be useful for targeted reduction of expression
levels.

The minimisation of metabolic adjustment (MOMA) method relaxes the
assumption of optimal growth flux for gene deletions by solving a
quadratic program to optimise distance minimisation in flux space
[@segre2002analysis]. This is because the minimal response to the
perturbation is considered to be a more accurate estimate of the true
flux state of the mutant [@raval2013introduction]. Initially, the flux
distribution for the mutant remains as close as possible to optimal flux
for the wild-type and deviates to form a suboptimal flux distribution
which is between that of the wild-type and mutant. In this way, MOMA is
able to predict phenotypic outcomes following knockouts more precisely
than FBA. Using a mechanistic model of reaction rates, integrative omics
metabolic analysis (IOMA) also solves a quadratic problem to deliver
kinetically-derived estimations of flux following genetic perturbation
[@yizhak2010integrating]. This is possible through integrating
quantitative proteomic and metabolomic data into the model, which
improves performance when compared to MOMA. Similarly, regulatory on/off
minimisation (ROOM) minimises the number of significant flux changes
following knockouts with respect to the wild type
[@shlomi2005regulatory]. This is achieved through redirecting flux
through alternative pathways following knockout. Many gene perturbation
experiments simulate knockouts under the assumption that there is no
downstream effect on gene regulation [@lee2012improving]. In the RELATCH
method, the principle of relative optimality is applied to predict how
cells adapt to perturbations by minimising relative flux patterns and
latent pathway activation with respect to a reference flux distribution
[@kim2012relatch]. As strains adapt to their perturbed state, they
undergo regulatory and metabolic changes, represented by two parameters
- one which penalises latent pathway activation and another which limits
enzyme contribution increases in active pathways. Unconventionally, the
REMEP method considers the impact of perturbations on metabolite as well
as flux patterns [@oyetunde2017metabolite]. This leads to improved flux
predictions for knockout mutants as the structure of cellular regulation
is represented more accurately.

It can be difficult to define the single most important objective in a
biological system as there are usually multiple conflicting cellular
objectives in addition to the maximisation of biomass, which is often
used as a proxy for growth. Methods such as Bayesian objective function
discrimination can be used for selection of the most suitable objective
function by using a probabilistic approach to compare multiple
objectives [@knorr2007bayesian]. Alternatively, there are methods which
utilise a weighted sum of objectives to scalarise multiple objectives
[@xu2011iterative]; however, it can be difficult to select weights that
elicit a uniform distribution of Pareto solutions and find solutions in
non-convex regions [@de2014global]. Thus, multi-objective optimisation
arguably presents the most realistic representation of metabolic flux in
biological systems by considering the contribution of a wide range of
competing objectives. The following section describes the main methods
used to implement multi-objective optimisation in metabolic models.
