--- 
title: "Metabolic Modelling Techniques"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
output: bookdown::gitbook
documentclass: book
bibliography: [refs.bib, packages.bib]
biblio-style: apalike
link-citations: yes
---
# Overview

```{r echo=FALSE, width=8, height=12}
temp <- purrr::quietly(library)(tidyverse)
temp <- purrr::quietly(library)(yaml)
temp <- purrr::quietly(library)(visNetwork)

treefun_character <- function(x){
  return(list(name = x, children = list()))
}

treefun_onelist <- function(x){
  list(name = names(x),
       children = treefun(x[[1]]))
}

treefun <- function(x){
  c(x %>% keep(is_list) %>% map(treefun_onelist),
    x %>% keep(is_character) %>% map(treefun_character))
}

getNodes <- function(x){
  bind_rows(tibble::as_data_frame(x[names(x)!='children']) %>%
              mutate(nchildren = length(x$children)), 
            map_df(x$children %>% keep(is_list), getNodes)
  )
}

getEdges <- function(x){
  if(length(x$children)==0){return(NULL)}
  
  bind_rows(data_frame(from = x$name, to = x$children %>% map_chr('name')),
            map_df(x$children, getEdges))
}

tax <- read_file('data/taxonomy.yaml') %>%
  yaml.load() %>%
  treefun_onelist()

visNetwork(nodes = getNodes(tax) %>% 
             mutate(id=name, 
                    title=name, 
                    label=name,
                    shape = if_else(nchildren>=1, 'box', 'text')),
           edges = getEdges(tax)) %>%
  visHierarchicalLayout(direction = 'LR', sortMethod = 'directed', levelSeparation=1000) %>%
  visPhysics(solver='hierarchicalRepulsion', 
             hierarchicalRepulsion = list(nodeDistance = 150, 
                                          springLength = 150, 
                                          springConstant= 0.01, 
                                          damping=0.5
             )
  ) %>%
  visNodes(font = list(size = 60, face = 'arial'), color = list(background = 'white')) %>%
  visEdges(smooth = list(enabled=TRUE, type='continuous')) %>%
  visExport(type='pdf')

```

## Abstract

Given the wealth of new methods for metabolic modelling, it is not difficult for a modeller to experience the situation of not being able to see the wood (or the forest) for the trees. Driven by this analogy, we here present the ``forest'' of principal methods used for constraint-based modelling in systems biology. We believe that the large scale comparison of the metabolic response of an organism to different environmental conditions will be an important challenge for genome-scale models. Therefore, within the context of multi-omic methods, we describe a tutorial for bacterial multi response (microarray data) to environmental conditions (including antibiotics, temperatures, heat shock). The tutorial for multi-objective optimisation uses the metabolic and transcriptomics adaptation estimator (METRADE) in MATLAB. Finally, we discuss potential implementations of machine-learning in metabolic modelling, towards automatically extracting knowledge from such models. Our updated classification of existing methods and existing tools allows us to suggest the most promising ones, with the aim to develop a vision of how existing methods could hybridise and become more complex.

